src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -1617,11 +1620,11 @@ public bool MoveNext()
                         }
                     }
                     _index = _dictionary._count + 1;
-                    _currentValue = default!; // TODO-NULLABLE-GENERIC
+                    _currentValue = default!; // TODO-NULLABLE: Remove ! when nullable attributes are respected
                     return false;
                 }
 
-                public TValue Current => _currentValue;
+                public TValue Current => _currentValue!;
```

Comment:
Should the return type be annotated as `MaybeNull`?

Reply:
Disregard, I remember this will depend on what generic parameter type is passed down to the Dictionary.

Reply:
Exactly.